odoo/tools/profiler.py

```diff
@@ -272,6 +272,94 @@ def post_process(self):
         super().post_process()
 
 
+class QwebCollector(Collector):
+    """
+    Record qweb execution with directive trace.
+    """
+    name = 'qweb'
+
+    def __init__(self):
+        super().__init__()
+        self.archs = {}
+        self.results = []
+        self.stack = []
+
+        def hook(event, env, **kwargs):
+            if event == 'begin':
+                self._handle_begin(env, **kwargs)
+            elif event == 'end':
+                self._handle_end(env, **kwargs)
+        self.hook = hook
+
+    def _handle_begin(self, env, view_id, xpath, directive, attrib, arch):
+        self.archs[view_id] = arch
```

Comment:
I would suggest to add a hook when starting rendering to inform about the arch. Not sure it is best solution yet, just an idea.